src/mctp-req.c

```diff
@@ -81,15 +81,20 @@ static int mctp_req(unsigned int net, mctp_eid_t eid,
 	if (rc != (int)len)
 		err(EXIT_FAILURE, "sendto(%zd)", len);
 
+	len = recvfrom(sd, NULL, 0, MSG_PEEK | MSG_TRUNC, NULL, 0);
+	if (len < 0)
```

Comment:
`len` is unsigned, this will never catch the error from `recvfrom()`. I'd suggest using a separate `ssize_t rlen` variable.

You'll probably want to handle `0` too, otherwise the `realloc()` below will (non-portably) turn into a `free`.

Reply:
Fixed